src/spaceone/dashboard/service/domain_dashboard_service.py

```diff
@@ -81,29 +82,16 @@ def update(self, params):
 
         domain_dashboard_vo: DomainDashboard = self.domain_dashboard_mgr.get_domain_dashboard(domain_dashboard_id,
                                                                                               domain_id)
-        version_vo = None
 
         if domain_dashboard_vo.viewers == 'PRIVATE' and \
                 domain_dashboard_vo.user_id != self.transaction.get_meta('user_id'):
             raise ERROR_PERMISSION_DENIED()
 
         if 'settings' in params:
-            if 'date_range' in params['settings']:
-                params['settings']['date_range'] = {
-                    'enabled': params['settings']['date_range'].get('enabled', False)
-                }
-            else:
-                params['settings'].update({'date_range': {'enabled': domain_dashboard_vo.settings.date_range.enabled}})
-
-            if 'currency' in params['settings']:
-                params['settings']['currency'] = {
-                    'enabled': params['settings']['currency'].get('enabled', False)
-                }
-            else:
-                params['settings'].update({'currency': {'enabled': domain_dashboard_vo.settings.currency.enabled}})
+            self._create_settings_in_params(domain_dashboard_vo, params['settings'])
```

Comment:
Clearly declare the results of function execution.
params['settings'] = self._create_settings_in_params(domain_dashboard_vo, params['settings'])